### ISSUE TITLE
Using more than one 'name' argument for on()?

### ISSUE THREAD

**Issue by shahreyar-abeer** [ColinFay/gargoyle#3]
`r 
library(shiny)
library(gargoyle)
options("gargoyle.talkative" = TRUE)
ui <- function(request){
  tagList(
    h4('Go'),
    actionButton("y", "y"),
    actionButton("z", "z"),
    h4('Output of z$v'),
    tableOutput("evt")
  )
}

server <- function(input, output, session){
  
  # Initiating the flags
  init( "airquality", "iris", "renderiris", "a2")
  
  # Creating a new env to store values, instead of
  # a reactive structure
  z <- new.env()
  
  observeEvent( input$y , {
    z$v <- mtcars
    # Triggering the flag
    trigger("airquality")
  })
  
  observeEvent( input$y , {
    #z$v <- mtcars
    # Triggering the flag
    trigger("a2")
  })
  
  on("airquality", {
    # Triggering the flag
    z$v <- airquality
    trigger("iris")
  })
  
  on("iris", {
    # Triggering the flag
    z$v <- iris
    trigger("renderiris")
  })
  
  ## need to use both 'iris' & 'a2'
  on({"iris", "a2"},{
    print("observed!")
  })
  
  output$evt <- renderTable({
    # This part will only render when the renderiris
    # flag is triggered
    watch("renderiris")
    head(z$v) 
  })
  
}

shinyApp(ui, server)
`

Note: It doesn't work!

**Comment by DavidJesse21** [comment]
Hi, I'm not one of the package authors but since the functionality is actually quite simple, I think I'm still able to answer your "question".

Tl;dr:
 For the functionality that you wish there won't be a solution that the package can provide (or at least not a simple one) and you might need to find your own way to accomplish your desired behaviour.

If you look at the source code of the `init` function:
https://github.com/ColinFay/gargoyle/blob/e0398a6970c5e43e081472e594fc13d1704bd938/R/funs.R#L66-L74

you can actually see that it does nothing else than adding a reactive value to session$userData for each "flag" that you want to initialize.
Now, `session$userData` under the hood is an environment.
However, you cannot subset an environment which would be kind of required for your wish, but only extract single elements out of it which is what the `{gargoyle}` functions do and thereby "trigger" the events, when the reactive values in session$userData change.

And to be honest I don't really get the scenario that you have in mind, in which you need this functionality.
In the example you have given both actions are triggered by input$y, so you should be just fine with leaving out `a2`.

Hope I could help you.

Best Regards
David

**Comment by christiansegercrantz** [comment]
Hey,

I'm also interested in the feature, not by "and" but by "or". Shouldn't it be possible to see if either of the triggers change and based on that run the expression? I'm sorry if you explained this already but I at least couldn't grasp it from your explanation.

Br,
Christian

**Comment by ilyaZar** [comment]
There may be more cleaner ways to provide the or-case as requested by the previous comment, but this one should work for the example of the initial PR comment; which is slightly modified so that a click on the `z`-button gives the trigger via `a2` and a click on the `y`-button the previous triggering-chain:

```r
library(shiny)
library(gargoyle)

options("gargoyle.talkative" = TRUE)
ui <- function(request){
  tagList(
    h4('Go'),
    actionButton("y", "y"),
    actionButton("z", "z"),
    h4('Output of z$v'),
    tableOutput("evt")
  )
}

server <- function(input, output, session){

  # Initiating the flags
  init( "airquality", "iris", "renderiris", "a2")

  # Creating a new env to store values, instead of
  # a reactive structure
  z <- new.env()

  observeEvent(input$y , {
    z$v <- mtcars
    # Triggering the flag
    trigger("airquality")
  })

  observeEvent(input$z , {
    #z$v <- mtcars
    # Triggering the flag
    trigger("a2")
  })

  on("airquality", {
    # Triggering the flag
    z$v <- airquality
    trigger("iris")
  })

  on("iris", {
    # Triggering the flag
    z$v <- iris
    trigger("renderiris")
  })

  ## need to use both 'iris' & 'a2'
  on(c("iris", "a2"),{
    print("observed!")
  })

  output$evt <- renderTable({
    # This part will only render when the renderiris
    # flag is triggered
    watch("renderiris")
    head(z$v)
  })

}
shinyApp(ui, server)
```
